packages/nfts/contracts/profile/RegisterProfilePicture.sol

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/ERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import { Ownable2StepUpgradeable } from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+
+contract RegisterProfilePicture is Initializable, UUPSUpgradeable, Ownable2StepUpgradeable {
+
+    error InvalidNFTContract(address nftContract);
+    error NotTokenOwner(address nftContract, uint256 tokenId, address caller);
+
+    // Mapping from user address to the selected NFT (Collection address and token ID)
+    struct ProfilePicture {
+        address nftContract;
+        uint256 tokenId;
+    }
+
+    mapping(address => ProfilePicture) public profilePictures;
+
+    event ProfilePictureSet(address indexed user, address indexed nftContract, uint256 indexed tokenId);
+
+    function initialize() public initializer {}
```

Comment:
Needs to initialize inherited contracts, and assign an owner/admin. The other contracts in this package are a good reference